src/rpcq-python.lisp

```diff
@@ -0,0 +1,287 @@
+                                 ("    \"\"\"~a\"\"\""   ,documentation))))
+                 (push (list deprecates slot-name required) deprecated-fields))
+                ;; recipe for a slot with a default value
+                (defaultp
+                 (python-out `(("    ~a: ~a = ~a"       ,(symbol-name slot-name)
+                                                        ,(python-maybe-optional-typing-type type required)
+                                                        ,(python-argspec-default type default))
+                               ("    \"\"\"~a\"\"\""    ,documentation))))
+                ;; recipe for a slot otherwise
+                (t
+                 (python-out `(("    ~a: ~a"           ,(symbol-name slot-name)
+                                                       ,(python-maybe-optional-typing-type type required))
+                               ("    \"\"\"~a\"\"\""   ,documentation)))))))
+          
+          ;; deprecated fields need special care.
+          (when deprecated-fields
+            ;; (1) add fake getters / setters
+            (dolist (field-spec deprecated-fields)
+              (destructuring-bind (old new new-required) field-spec
+                (declare (ignore new-required))
+                (when new
+                  (python-out `(("    @property")
+                                ("    def ~a(self):"                                ,(symbol-name old))
+                                ("        warn('~a is deprecated, use ~a instead')" ,(symbol-name old)
+                                                                                    ,(symbol-name new))
+                                ("        return self.~a"                           ,(symbol-name new))
+                                ("")
+                                ("    @~a.setter"                                   ,(symbol-name old))
+                                ("    def ~a(self, value):"                         ,(symbol-name old))
+                                ("        warn('~a is deprecated, use ~a instead')" ,(symbol-name old)
+                                                                                    ,(symbol-name new))
+                                ("        self.~a = value"                          ,(symbol-name new)))))))
+            
+            ;; (2) add extra fields to output
+            (python-out `(("    def _extend_by_deprecated_fields(self, d):")
+                          ("        super()._extend_by_deprecated_fields(d)")))
+            (dolist (field-spec deprecated-fields)
+              (destructuring-bind (old new new-required) field-spec
+                (declare (ignore new-required))
+                (when new
+                  (python-out `(("        d.~a = d.~a" ,(symbol-name old)
+                                                       ,(symbol-name new)))))))
+            
+            ;; (3) tolerate extra fields on input
+            (format stream
```

Comment:
use `python-out` here?

Reply:
I switched to `format` here b/c `python-out` inserts some newlines automatically, and they're great everywhere else but they make the output here look ugly.